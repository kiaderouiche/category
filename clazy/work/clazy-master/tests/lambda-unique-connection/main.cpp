#include <QtCore/QObject>

void globalFunc();

void test()
{
    QObject *o;
    o->connect(o, &QObject::destroyed, o, []{}, Qt::UniqueConnection); // Warn
    o->connect(o, &QObject::destroyed, o, []{}, Qt::ConnectionType(Qt::UniqueConnection | Qt::DirectConnection)); // Warn
    o->connect(o, &QObject::destroyed, o, []{}); // OK
    o->connect(o, &QObject::destroyed, o, globalFunc); // OK
    QObject::connect(o, &QObject::destroyed, o, []{}, Qt::UniqueConnection); // Warn
    o->connect(o, &QObject::destroyed, o, globalFunc, Qt::UniqueConnection); // Warn
    o->connect(o, &QObject::destroyed, o, &QObject::deleteLater, Qt::UniqueConnection); // OK
    o->connect(o, QOverload<QObject*>::of(&QObject::destroyed), o, &QObject::deleteLater, Qt::UniqueConnection); // OK
    o->connect(o, &QObject::destroyed, o, QOverload<>::of(&QObject::deleteLater), Qt::UniqueConnection); // OK
}

