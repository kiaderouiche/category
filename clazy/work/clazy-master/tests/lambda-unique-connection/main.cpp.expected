lambda-unique-connection/main.cpp:8:49: warning: UniqueConnection is not supported with non-member functions [-Wclazy-lambda-unique-connection]
lambda-unique-connection/main.cpp:9:49: warning: UniqueConnection is not supported with non-member functions [-Wclazy-lambda-unique-connection]
lambda-unique-connection/main.cpp:12:55: warning: UniqueConnection is not supported with non-member functions [-Wclazy-lambda-unique-connection]
lambda-unique-connection/main.cpp:13:55: warning: UniqueConnection is not supported with non-member functions [-Wclazy-lambda-unique-connection]
